File: 30_days_of_code/day_08.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
import sys

count, *data = sys.stdin.read().splitlines()
phone_book = {}

for i in range(int(count)):
    name, phone = data[i].split(" ")
    phone_book[name] = phone
    
for i in range(int(count), len(data)):
    name = data[i]
    
    if (name in phone_book):
        print(f"{name}={phone_book[name]}")
    else:
        print("Not found")
        

File: hackerrank/30_days_of_code/day08.js
function processData(input) {
    //Enter your code here
    const [count, ...data] = input.split("\n")
    let phoneBook = {}
    
    for (let i = 0; i < count; i++){
        let [name, phone] = data[i].split(" ")
        phoneBook[name] = phone
    }
    
    for (let i = count; i < data.length; i++){
        let name = data[i];
        
        if (name in phoneBook){
            console.log(`${name}=${phoneBook[name]}`);
        } else {
            console.log("Not found");
        }
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
